lib/CMakeLists.txt

```
@@ -19,4 +19,20 @@ if(ASYNC_MQTT_USE_LOG)
 endif()
 target_compile_definitions(async_mqtt_separate PRIVATE ASYNC_MQTT_SEPARATE_COMPILATION)
 
+macro(fix_msvc_build)
+    message(STATUS "MSVC build fix applied")
+    target_compile_options(async_mqtt_separate PRIVATE "/Zc:preprocessor" "/bigobj")
+endmacro()
+
+get_property(is_multi_config GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+if(is_multi_config)
+    if ($<STREQUAL:$<CXX_COMPILER_ID>,"MSVC">)
```

```suggestion
    if (MSVC)
```

async_mqtt uses `MSVC` at other place. So if you can use it here, please use it.
See https://cmake.org/cmake/help/latest/variable/MSVC.html

---

Done